zenoh/src/net/routing/interceptor/access_control.rs

```diff
@@ -429,14 +473,6 @@ impl InterceptorTrait for EgressAclEnforcer {
             | NetworkBody::Declare(Declare {
                 body: DeclareBody::UndeclareToken(_),
                 ..
-            })
-            | NetworkBody::Declare(Declare {
-                body: DeclareBody::UndeclareQueryable(_),
-                ..
-            })
-            | NetworkBody::Declare(Declare {
-                body: DeclareBody::UndeclareSubscriber(_),
-                ..
             }) => {}
             // Unfiltered remaining message types
             NetworkBody::Interest(_) | NetworkBody::OAM(_) | NetworkBody::ResponseFinal(_) => {}
```

Comment:
If `Request` and `Response` are handled, then `ResponseFinal` should be handled too.

Reply:
`ResponseFinal` Cannot and should not be filtered:
- It cannot be filtered because it does not have a keyexpr
- It should not be filtered because it should always be allowed, even if all responses are denied the `ResponseFinal` is allowed through to mark the end of the query.

Reply:
> It cannot be filtered because it does not have a keyexpr

If at least one `Response` keyexpr is denied, then all `ResponseFinal` would be denied.

> It should not be filtered because it should always be allowed, even if all responses are denied the ResponseFinal is allowed through to mark the end of the query.

Queries are cleaned up after a timeout, the query sender doesn't need `ResponseFinal` to finalize them. If I understand correctly, not filtering the `ResponseFinal` would return an empty reply rather than error. If I misconfigure my ACL rules w.r.t queries, I would expect my queries to return errors and not succeed with empty result sets.

Reply:
> If I misconfigure my ACL rules w.r.t queries, I would expect my queries to return errors and not succeed with empty result sets.

ACL should give the illusion that the denied queryable replies do not exist in the network, which is the current behavior of "succeed with empty result sets" (i.e. a set that only contains the allowed replies). An error when replies are denied would leak the existence of the replies and the respective ACL config that is denying them in the network.